src/dbnode/persist/fs/files.go

```diff
@@ -758,9 +758,11 @@ func forEachInfoFile(
 		if err != nil {
 			continue
 		}
-		if len(matched[i].AbsoluteFilepaths) != 1 {
-			continue
-		}
+		// Sort the matched fileset files so info file comes first.
```

Comment:
Hm this seems like a weird contract to offer.

Can we simply add a method to `FileSetFile` that returns the info file path if it exists or not in absolute file paths:
```go
func (f *FileSetFile) InfoFilePath() (string, bool) {
  // match against f.AbsoluteFilePaths and
  if found {
    return f.AbsoluteFilePaths[foundIdx], true
  }
  return "", false
}
```